csrc/scheduler/normalization_utils.cpp

```diff
@@ -827,8 +860,17 @@ bool projectBufferToInputs(
     return false;
   }
 
+  // must project to inputs otherwise don't have enough register or shared
+  // memory to store the buffers
+  int64_t max_available_buffer = getMaxRegOrSharedMemorySizeForPersistentBuffer(
+      runtime_info, persistent_buffer_info.persistent_buffers);
+  if (max_available_buffer <
+      persistent_buffer_size_info.persistent_buffer_size) {
+    return true;
```

Comment:
Could you add `NVF_ERROR(max_available_buffer >= persistent_buffer_size_info.projected_persistent_buffer_size)`?